src/main/java/org/apache/cassandra/sidecar/metrics/FilteringMetricRegistry.java

```diff
@@ -0,0 +1,226 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sidecar.metrics;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Predicate;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.NoopMetricRegistry;
+import com.codahale.metrics.Timer;
+import io.vertx.core.impl.ConcurrentHashSet;
+
+/**
+ * Allows filtering of metrics based on configured allow list. Metrics are filtered out before registering them.
+ */
+public class FilteringMetricRegistry extends MetricRegistry
+{
+    private static final NoopMetricRegistry NO_OP_METRIC_REGISTRY = new NoopMetricRegistry(); // supplies no-op metrics
+    private final Predicate<String> isAllowed;
```

Comment:
Maybe call it predicate?
```suggestion
    private final Predicate<String> isAllowedPredicate;
```

Reply:
I feel adding `Predicate` to name is redundant. We don't add that for other members

Reply:
I don't feel the variable name fully conveys what the variable is supposed to do. Maybe another name is possible too?

Reply:
For readability added `Predicate` extension